python/lsst/pipe/tasks/calexpCutout.py

```diff
@@ -0,0 +1,113 @@
+import lsst.pipe.base as pipeBase
+import lsst.geom as geom
+from lsst.pex.config import Field
+from lsst.meas.algorithms import Stamp, Stamps
+
+__all__ = ['CalexpCutoutTaskConfig', 'CalexpCutoutTask']
+DETECTOR_DIMENSIONS = ('instrument', 'visit', 'detector')
+
+
+class CalexpCutoutTaskConnections(pipeBase.PipelineTaskConnections,
+                                  dimensions=DETECTOR_DIMENSIONS,
+                                  defaultTemplates={}):
+    """Connections class for CalexpCutoutTask
+    """
+    in_table = pipeBase.connectionTypes.Input(
+        doc="Locations for cutouts",
+        name="cutout_positions",
+        storageClass="AstropyQTable",
+        dimensions=DETECTOR_DIMENSIONS,
+    )
+    calexp = pipeBase.connectionTypes.Input(
+        doc="Calexp objects",
+        name="calexp",
+        storageClass="ExposureF",
+        dimensions=DETECTOR_DIMENSIONS,
+    )
+    cutouts = pipeBase.connectionTypes.Output(
+        doc="Cutouts",
+        name="calexp_cutouts",
+        storageClass="Stamps",
+        dimensions=DETECTOR_DIMENSIONS,
+    )
+
+
+class CalexpCutoutTaskConfig(pipeBase.PipelineTaskConfig,
+                             pipelineConnections=CalexpCutoutTaskConnections):
+    """Configuration for CalexpCutoutTask
+    """
+    max_cutouts = Field(dtype=int, default=100, doc='Maximum number of entries to process. '
+                                                    'The result will be the first N in the input table.')
+    skip_bad = Field(dtype=bool, default=True, doc='Skip cutouts that do not fall completely within'
+                                                   ' the calexp bounding box?  If set to False a ValueError'
+                                                   ' is raised instead.')
+
+
+class CalexpCutoutTask(pipeBase.PipelineTask):
+    """Task for computing cutouts on a specific calexp given
+    positions and sizes of the stamps.
+    """
+    ConfigClass = CalexpCutoutTaskConfig
+    _DefaultName = "calexpCutoutTask"
+
+    def run(self, in_table, calexp):
+        """Compute and return the cutouts.
+
+        Parameters
+        ----------
+        in_table : `astropy.QTable`
+            A table containing at least the following columns: position, size.
+            The position should be an `astropy.SkyCoord`.  The size is in pixels.
+        calexp : `lsst.afw.image.ExposureF`
+            The calibrated exposure from which to extract cutouts
+
+        Returns
+        -------
+        output : `lsst.pipe.base.Struct`
+            A struct containing a `lsst.meas.algorithms.Stamps` object
+            that wraps a list of masked images of the cutouts and a
+            `PropertyList` containing the metadata to be persisted
+            with the cutouts.  The exposure metadata is preserved and,
+            in addition, arrays holding the RA and Dec of each stamp
+            in degrees are added to the metadata.
```

Comment:
I strongly suggest giving the field names (the user should not have to read the code to find these). For example:
```
A struct containing:

* cutouts: an `lsst.meas.algorithms.Stamps` object that wraps ...
* metadata: an `lsst.daf.base.PropertyList` containing ...
```